File: AppSets-Android/appsets/src/main/kotlin/xcj/app/appsets/settings/AppConfig.kt
package xcj.app.appsets.settings

data class AppConfiguration(
    val canSignUp: Boolean,
    val apiSchema: String,
    val apiHost: String,
    val apiPort: Int,
    val apiUrl: String,
    val appsetsAppId: String,
    val imBrokerProperties: String
)

object AppConfig {
    private var updateTime: Long = 0
    private var updating = false

    var isTest: Boolean = true

    var appConfiguration: AppConfiguration = createAppConfiguration()

    private fun createAppConfiguration(): AppConfiguration {
        return AppConfiguration(
            canSignUp = true,
            apiSchema = "https",
            apiHost = "8.137.93.144",
            apiPort = 3401,
            apiUrl = "",
            appsetsAppId = "APPSETS2023071579019880338529",
            imBrokerProperties = ""
        )
        return if (isTest) {
            AppConfiguration(
                canSignUp = true,
                apiSchema = "https",
                apiHost = "127.0.0.1",
                apiPort = 8084,
                apiUrl = "",
                appsetsAppId = "APPSETS2023071579019880338529",
                imBrokerProperties = ""
            )
        } else {
            AppConfiguration(
                canSignUp = true,
                apiSchema = "https",
                apiHost = "8.137.93.144",
                apiPort = 3401,
                apiUrl = "",
                appsetsAppId = "APPSETS2023071579019880338529",
                imBrokerProperties = ""
            )

        }
    }

    fun isNeedUpdateImBrokerProperties(): Boolean {
        return appConfiguration.imBrokerProperties.isEmpty()
    }

    fun updateImBrokerProperties(properties: String) {
        appConfiguration = appConfiguration.copy(imBrokerProperties = properties)
    }

    fun update() {
        if (updating) {
            return
        }
    }
}

File: AppSets-iOS/AppSets/com/app/appsets/settings/AppConfig.swift
//
//  AppConfig.swift
//  AppSets"
//
//  Created by caiju Xu on 2024/12/20.
//

class AppConfiguration{
    var canSignUp:Bool
    var apiSchema:String
    var apiHost:String
    var apiPort:Int
    var apiUrl:String
    var appSetsAppId:String
    var imBrokerProperties:String
    
    init(canSignUp: Bool, apiSchema: String, apiHost: String, apiPort: Int, apiUrl: String, appSetsAppId: String, imBrokerProperties: String) {
        self.canSignUp = canSignUp
        self.apiSchema = apiSchema
        self.apiHost = apiHost
        self.apiPort = apiPort
        self.apiUrl = apiUrl
        self.appSetsAppId = appSetsAppId
        self.imBrokerProperties = imBrokerProperties
    }
    
}

class AppConfig {
    
    private static let TAG = "AppConfig"
    
    public static let Instance: AppConfig = AppConfig()
    
    private var updateTime: Int = 0
    
    private var updating: Bool = false
    
    var appConfiguration: AppConfiguration = AppConfiguration(
        canSignUp: true,
        apiSchema: "https",
        apiHost: "192.168.15.184",
        apiPort: 8084,
        apiUrl: "",
        appSetsAppId: "APPSETS2023071579019880338529",
        imBrokerProperties: ""
    )
    
    private init(){
        
    }
    
    
    func updateIMBrokerProperties(properties:String){
        PurpleLogger.current.d(AppConfig.TAG, "updateIMBrokerProperties")
        appConfiguration.imBrokerProperties = properties
    }
}
